### DISCUSSION TITLE
Is it feasible to use Layer2 to improve tps?

### DISCUSSION THREAD

**Discussion by Julian332** [ConsenSys/quorum#1553] [Q&A] [answered]
like title

**Comment by baptiste-b-pegasys** [comment, marked as answer]
it is as feasible as in testnet and mainnet for public txs.

I have not tested it, but took a look at it.

There are many things to take in account:
- type of L2: optimistic rollup, zkEVM etc
- GoQ's private transactions might not be useful anymore, or maybe for some specific usage you still may want in parallel of L2.
- private transactions with a L2 is something to be designed, there is no L2 fully compatible with GoQ
- migration of contracts: if contracts are not designed to be bridged, I think that only a dump+restore is feasible
- why having GoQuorum as L1 for a L2 you will mainly use? (consensus protocol?, enterprise features?)

**Reply by Julian332** [reply]
thanks , BFT consensus and highe tps is the main reason
